ipc/nano/src/lib.rs

```diff
@@ -54,9 +54,9 @@ impl<S> GuardedSocket<S> where S: WithSocket<Socket> {
 }
 
 impl<S> Deref for GuardedSocket<S> where S: WithSocket<Socket> {
-    type Target = S;
+    type Target = Arc<S>;
 
-    fn deref(&self) -> &S {
+    fn deref(&self) -> &Arc<S> {
```

Comment:
@rphmeier because actual dispatch is implemented for Arc<T>, but not for T

see here:
https://github.com/ethcore/parity/blob/master/ipc/codegen/src/codegen.rs#L838
```
		impl $host_generics ::ipc::IpcInterface<$interface_endpoint> for ::std::sync::Arc<$interface_endpoint> $where_clause {
}
```